pkg/types/json_binary_functions.go

```diff
@@ -153,15 +164,21 @@ func unquoteJSONString(s string) (string, error) {
 // According RFC 3629, the max length of utf8 characters is 4 bytes.
 // And MySQL use 4 bytes to represent the unicode which must be in [0, 65536).
 func decodeEscapedUnicode(s []byte) (char [4]byte, size int, err error) {
-	size, err = hex.Decode(char[0:2], s)
-	if err != nil || size != 2 {
-		// The unicode must can be represented in 2 bytes.
+	size, err = hex.Decode(char[0:4], s)
+	if err != nil || (size != 2 && size != 4) {
```

Comment:
when `(size != 2 && size != 4)` happens, `err` is nil, caller's check about `err == nil` will go wrong. I think we should return an error here?